### PR TITLE
Set ManagedDependenciesPath for both AppService and Linux Consumption environments

### LINKED ISSUE #632: Set ManagedDependenciesPath for both AppService and Linux Consumption environments
Follow the same logic as the Functions Host. For more information, see https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Environment/EnvironmentExtensions.cs#L240.

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
Resolves https://github.com/Azure/azure-functions-powershell-worker/issues/632.

### CODE REVIEW

File: test/Unit/DependencyManagement/DependencyManagementTests.cs [hunk 1]
```diff
@@ -449,6 +449,62 @@ public void FunctionAppExecutionShouldContinueIfPreviousDependenciesExist()
...
+                    Directory.CreateDirectory(functionAppRoot);
+                }
+
+                var managedDependenciesPath = ManagedDependenciesPathDetector.GetManagedDependenciesPath(functionAppRoot);
+                var dependenciesPathIsValid = managedDependenciesPath.StartsWith(expectedPath, StringComparison.CurrentCultureIgnoreCase);
+                Assert.True(dependenciesPathIsValid);
+            }
+            finally
+            {
+                if (deleteVariable)
```
Comment (AnatoliB) [inline test/Unit/DependencyManagement/DependencyManagementTests.cs]: If you just use `setEnvironmentVariable` here, you don't seem to need `deleteVariable` anymore

File: test/Unit/DependencyManagement/DependencyManagementTests.cs [hunk 2]
```diff
@@ -449,6 +449,62 @@ public void FunctionAppExecutionShouldContinueIfPreviousDependenciesExist()
...
+            {
+                string appDataFolder = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData, Environment.SpecialFolderOption.DoNotVerify);
+                expectedPath =  Path.Combine(appDataFolder, AzureFunctionsFolderName, "MyFunction", ManagedDependenciesFolderName);
+            }
+
+            var functionAppRoot = Path.Join(Path.GetTempPath(), "MyFunction");
+
+            try
+            {
+                if (!Directory.Exists(functionAppRoot))
```
Comment (AnatoliB) [inline test/Unit/DependencyManagement/DependencyManagementTests.cs]: No need for this check: `Directory.CreateDirectory` will not fail if the directory already exists

File: test/Unit/DependencyManagement/DependencyManagementTests.cs [hunk 3]
```diff
@@ -449,6 +449,62 @@ public void FunctionAppExecutionShouldContinueIfPreviousDependenciesExist()
                 TestCaseCleanup();
             }
         }
+
+        [Theory]
+        [InlineData("CONTAINER_NAME", "MyContainerName", true)]
+        [InlineData("WEBSITE_INSTANCE_ID", "MyInstanceId", true)]
+        [InlineData(null, null, false)]
+        public void ValidateManagedDependenciesPath(string name, string value, bool setEnvironmentVariable)
```
Comment (AnatoliB) [inline test/Unit/DependencyManagement/DependencyManagementTests.cs]: Consider moving this test to a separate file: this looks very much like a unit test for the ManagedDependenciesPathDetector class

**Review by AnatoliB — APPROVED** [review]
LGTM with some minor comments

File: test/Unit/DependencyManagement/DependencyManagementTests.cs (hunk 2 above)
Comment (Francisco-Gamino) [reply test/Unit/DependencyManagement/DependencyManagementTests.cs]: Fixed.

File: test/Unit/DependencyManagement/DependencyManagementTests.cs (hunk 3 above)
Comment (Francisco-Gamino) [reply test/Unit/DependencyManagement/DependencyManagementTests.cs]: Great suggestion. Fixed.

File: test/Unit/DependencyManagement/DependencyManagementTests.cs (hunk 1 above)
Comment (Francisco-Gamino) [reply test/Unit/DependencyManagement/DependencyManagementTests.cs]: Fixed.

**Review by AnatoliB — APPROVED** [review]
LGTM, thank you

**Pull request merged**